osu.Game/Screens/Play/HUD/PerformancePointsCounter.cs

```diff
@@ -133,10 +136,12 @@ private void onJudgementChanged(JudgementResult judgement)
         [CanBeNull]
         private DifficultyAttributes getAttributeAtTime(JudgementResult judgement)
         {
-            if (timedAttributes == null || timedAttributes.Count == 0)
+            if (timedAttributes == null || timedAttributes.Count == 0 || gameplayClock == null)
                 return null;
 
-            int attribIndex = timedAttributes.BinarySearch(new TimedDifficultyAttributes(judgement.HitObject.GetEndTime(), null));
+            double judgementTime = judgement.HitObject.GetEndTime() / gameplayClock.TrueGameplayRate;
```

Comment:
Is this implying that the attributes are all times in real-time? I think I'd want an inline comment here explaining this, because it looks like a huge WTF to my eyes even after reading the surrounding code.

Reply:
Yes, this is what's used to populate the attributes:
https://github.com/ppy/osu/blob/6f941a891618e22f6316a740a30e9d47a85f6c92/osu.Game/Rulesets/Difficulty/Preprocessing/DifficultyHitObject.cs#L33-L36

The other way of going about this would be to invert the above.

Reply:
That is mind boggling complexity there. Is there a reason for it being done that way? Feels like something we are going to trip over many times in the future, potentially introducing serious PP bugs.

If it stays this way, the variables should be renamed, and maybe duplicated with non-adjusted versions.

Reply:
In the difficulty calculation part of code? Because time deltas are very important. Actually, we _already had_ bugs because these properties weren't adjusted for clockrate which was fixed in https://github.com/ppy/osu/pull/11849

I'm definitely not about to change diffcalc.

Reply:
Are you against renaming them at least? It's very weird that comparing a `DifficultyHitObject.EndTime` with `HitObject.EndTime` for the *same object* will actually differ..

Reply:
I'm not, but it feels pointless because diffcalc should internally never ever deal with raw hitobject times. At the end of the day the issue is in what's exposed by `CalculateTimed`, which is still going to be an issue with or without the rename.

Reply:
Hmm okay.

I'm not sure what the intention of `CalculateTimed` is even, since there's no xmldoc, so I guess populating that would  be a good start. Not sure which direction you want to take this fix, but if the solution you currently have proposed is to stick, it definitely needs an inline comment explaining how it works.

Reply:
I've inverted the implementation to expose the non-clock-adjusted time from `TimedDifficultyAttributes` instead.